src/foam/core/session/PersistentSessionDAO.js

```diff
@@ -0,0 +1,165 @@
+/**
+ * @license
+ * Copyright 2025 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.core.session',
+  name: 'PersistentSessionDAO',
+  extends: 'foam.dao.ProxyDAO',
+  implements: [
+    'foam.core.COREService'
+  ],
+
+  documentation: `The key to 'persistent' sessions is the userId on the
+session. If a user makes a client call with an existing session id and
+a matching user id, then the system considers the user logged in.
+Sessions are NOT-frozen, for reasons discussed elsewhere.  The
+implication is updates are not persisted to the runtime journal because
+there is never a delta.
+This DAO maintains it's own copy of the session to calculate the delta
+and then makes an explicit journal put call to store the session with
+the user id.
+
+Additionally, anonymous also adds a complication.  When a user is logged
+out, the client will present anonymous views and the anonymous user
+uses the same session id as it is the same client connection. This DAO
+also ensures the session is not saved with the anonymous user id.
+`,
+
+  javaImports: [
+    'foam.core.auth.AuthService',
+    'foam.core.auth.ServiceProvider',
+    'foam.core.logger.Logger',
+    'foam.core.logger.Loggers',
+    'foam.dao.ArraySink',
+    'foam.dao.DAO',
+    'foam.dao.java.JDAO',
+    'foam.dao.Journal',
+    'foam.dao.MDAO',
+    'foam.dao.NullDAO',
+    'foam.dao.ProxyDAO',
+    'foam.lang.ContextAgent',
+    'foam.lang.ContextAgentTimerTask',
+    'foam.lang.X',
+    'java.util.HashMap',
+    'java.util.List',
+    'java.util.Map',
+    'java.util.Timer'
+  ],
+
+  javaCode: `
+  public PersistentSessionDAO(X x, DAO delegate) {
+    setX(x);
+    setDelegate(delegate);
+  }
+  `,
+
+  properties: [
+    {
+      name: 'persistentDAO',
+      class: 'foam.dao.DAOProperty',
+      javaFactory: 'return new MDAO(Session.getOwnClassInfo());'
+    },
+    {
+      name: 'mDAO',
+      class: 'foam.dao.DAOProperty'
+    },
+    {
+      name: 'journal',
+      class: 'FObjectProperty',
+      of: 'foam.dao.Journal'
+    },
+    {
+      name: 'anonymousCache',
```

Comment:
I assume the anonymous cache is for performance?  Can you add some documentation on why we have this?

Any reason the underlying mdao is not fast enough?